presto-main/src/main/java/com/facebook/presto/cost/FragmentedPlanStatsCalculator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.RemoteSourceNode;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.cost.PlanNodeStatsEstimateMath.addStatsAndMaxDistinctValues;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Verify.verify;
+
+public class FragmentedPlanStatsCalculator
```

Comment:
I don't think we still need support for the `CoefficientBasedStatsCalculator`, and getting rid of the `SelectingStatsCalculator` would be nice too.  But unless we want to get rid of the StatsCalculator interface so that the only kind of StatsCalculator is a ComposableStatsCalculator, it can't be a rule. Honestly I don't think that's a bad idea because the ComposableStatsCalculator is by definition super extensible and flexible. What do you think?

Reply:
Yeah. I think we can remove the `CoefficientBasedStatsCalculator`, and rename `ComposableStatsCalculator` to `StatsCalculator` removing the interface.

Reply:
https://github.com/prestodb/presto/pull/11286